guides/create_an_app/2-channels.md

```diff
@@ -1,5 +1,163 @@
+
+Once a user joins a channel they can send messages to the channel and they’ll be received by one of the `handle_in` messages. The first one just returns the payload sent by the user back to them. The second one sends the payload to all the users subscribed to the channel. It’s the second that allows us to build our chat room with just a little frontend work. When you send a message to the chat room, this function sends it back to you and to everyone else in the channel.
+
+To handle the connections on the Client-side we’ll start by adding **jQuery** to our app. First of all lets install it via npm command inside `assets/` directory:
+
+```console
+cd assets/
+```
+
+```console
+npm install jquery --save
+```
+
+Then we need to let our bundle know that we want to include newly installed library to final build. To achieve that we need to visit `assets/webpack.config.js` config file and add following.
+
+On top of the config file
+
+```javascript
+const Webpack = require('webpack');
+```
+
+Then we need to update `plugins` section one the bottom of this file. It should look like that:
+
+```javascript
+plugins: [
+  new MiniCssExtractPlugin({ filename: '../css/app.css' }),
+  new CopyWebpackPlugin([{ from: 'static/', to: '../' }]),
+  new Webpack.ProvidePlugin({
+    $: 'jquery',
+    jQuery: 'jquery',
+    'window.jQuery': 'jquery'
+  })
+]
+```
+
+Phoenix comes with a simple javascript socket client, but it’s disabled by default. Go into your `assets/js/app.js` and uncomment the last line:
+
+```javascript
+import socket from "./socket"
+```
+
+Then go into your `assets/js/socket.js` and paste the following:
+
+```javascript
+socket.connect()
+let channel = socket.channel("general:lobby", {});
+let list    = $('#message-list');
+let message = $('#message');
+let name    = $('#name');
+
+message.on('keypress', event => {
+  if (event.keyCode == 13) {
+    channel.push('shout', { name: name.val(), message: message.val() });
+    message.val('');
+  }
+});
+
+channel.on('shout', payload => {
+  list.append(`<b>${payload.name || 'Anonymous'}:</b> ${payload.message}<br>`);
+  list.prop({scrollTop: list.prop("scrollHeight")});
+});
+
+channel.join()
+  .receive("ok", resp => { console.log("Joined successfully", resp) })
+  .receive("error", resp => { console.log("Unable to join", resp) })
+```
+
+`socket.channel("random:lobby", {})` sends the join request to the server, and the server sends the message back to the client. Here, we listen for a keypress event on the message text field. Whenever the user enters a message, it’s pushed on the channel and the text field is cleared. When there’s an incoming message on the channel, it’s appended to the **div** we previously created and scrolled to the bottom.
+
+Next we will want to create the container for our messages. Open `lib/workshops_app_web/templates/page/index.html.eex`, and replace its contents with:
+
+```html
+<div id='message-list' class='row'>
+</div>
+
+<div class='row form-group'>
```

Comment:
I wonder if we really need this CSS classes, here. Maybe we might leave here, only required things, like `id`s of elements.

Reply:
Well, this is just to have name input and msg input just in one row. 🤔 Don't have an opinion on that if it looks better in a single line or in two lines.

Reply:
TBH, I think I will leave it as it is. With this class, it look more like a chat app.